File: 1st year/hotel project c++/projectcpp/src/Room_Type_C.cpp
#include "Room_Type_C.h"
#include<iostream>
Room_Type_C::Room_Type_C()
{
    //ctor
}

Room_Type_C::~Room_Type_C()
{
    //dtor
}
Room_Type_C:: Room_Type_C(int maximum_capacity, double price_per_person, int people, int days)

		: Room (maximum_capacity, price_per_person)//kanei gia thn megisth xwrithkothta kai gia thn timh ana atomo
		{
		    										  //oti kai o constructor ths Room
		minimum_people = people;//thetei ton elaxisto arithmo atomwn apo to orisma
		minimum_days = days;//thetei ton elaxisto arithmo hmerwn apo to orisma


	}
	bool Room_Type_C::new_booking(Booking* booketo) { //elexgei an ginetai kainouria krathsh kai thn kanei
	if (maximum_capacity < booketo->getPeople() || minimum_people>booketo->getPeople() || minimum_days<booketo->getDays_of_accomodation()) {
			//elegxei an threitai h megisth xwritikothta kai o elaxistos arithmos hmerwn kai atomwn
			return false;//epistrefei false an den threitai
		}
	if (maximum_capacity < booketo->getPeople()) {//elegxei an h xwrithkothta einai mikroterh apo ta atoma
		return false;
	} //an einai epistrefei false kai termatizei
	if(booketo->getArrival()+booketo->getDays_of_accomodation()-1<=30){
		for (int j = booketo->getArrival(); j <= (booketo->getArrival() + booketo->getDays_of_accomodation() - 1); j++)
			//gia j apo thn mera afiksh mexri thn hmera pou tha fygoun(hmera afikshs+ meres pou menoun)
			if (availability[j] != 0) {//an h diahesimothta ekeinh thn hmera den einai kenh (exei yparksei krathsh se auto to dwmatio gia autes ths meres dhladh
				return false;
			}//epistrefei false kai termatizei

		for (int j = booketo->getArrival(); j <= (booketo->getArrival() - 1 + booketo->getDays_of_accomodation()); j++)
			//to j einai idio me to j apo thn prohgoumenh for
			availability[j] =booketo; //bazei se kathe keli ths diathesimothtas tou dwmatiou ekeinou tou dwmatiouta stoixeia ths krathshs
		booketo->setRoom(this);//thetei thn anafora thw krathshs se auto to dwmatio
		return true;  //epistrefei true kai termatizei

	}
	else{
		cout << "Out of Range error: " <<endl;
	//uparxei periptwsh na dothoun meres pou den einai sta oria
														//tou pinaka diathesimothtas
		return false;//tote epistrefei false
	}}

File: 1st year/hotel project java/project/Room_Type_C.java
 
public class Room_Type_C extends Room{
	private int minimum_people,minimum_days;//elaxistos arithmos atomwn kai elaxistos arithmos hmerwn
	//constructor
	Room_Type_C(){
		
	}
	Room_Type_C(int maximum_capacity,double price_per_person,int people,int days)
	{  super(maximum_capacity,price_per_person);//kanei gia thn megisth xwrithkothta kai gia thn timh ana atomo 
	    //oti kai o constructor ths Room
	    this.minimum_people=people;//thetei ton elaxisto arithmo atomwn apo to orisma
	    this.minimum_days=days;//thetei ton elaxisto arithmo hmerwn apo to orisma
	   
	   
	   }
	   //methods
	public int getMinimum_people() {
		return minimum_people;
	}

	public void setMinimum_people(int minimum_people) {
		this.minimum_people = minimum_people;
	}

	public int getMinimum_days() {
		return minimum_days;
	}

	public void setMinimum_days(int minimum_days) {
		this.minimum_days = minimum_days;
	}
	
	boolean new_booking(Booking booketo)//yperkalyptei thn methodo ths roo  new_booking 
	{
			if(this.maximum_capacity < booketo.getPeople()&& this.minimum_people>booketo.getPeople()&&this.minimum_days<booketo.getDays_of_accomodation()){
            //elegxei an threitai h megisth xwritikothta kai o elaxistos arithmos hmerwn kai atomwn 
			    return false;//epistrefei false an den threitai
			 }                                     
        //alliws kanei oti kai h new_booking ths Room
			 try{
         for(int j = booketo.getArrival();j <=(booketo.getArrival() + booketo.getDays_of_accomodation()-1);j++) 
         if(this.availability[j] != null){
            return false;}                                                  
         
          for(int j = booketo.getArrival();j <=(booketo.getArrival()-1 + booketo.getDays_of_accomodation());j++) 
          this.availability[j] = booketo;      
          booketo.setRoom(this);
          return true;  
             
            }
        catch(java.lang.ArrayIndexOutOfBoundsException e){
            return false;
		}
		}
}
